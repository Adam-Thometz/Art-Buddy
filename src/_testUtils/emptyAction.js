const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION