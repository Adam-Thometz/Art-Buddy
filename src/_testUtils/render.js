import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
