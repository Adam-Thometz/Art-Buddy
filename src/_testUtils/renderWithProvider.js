import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;