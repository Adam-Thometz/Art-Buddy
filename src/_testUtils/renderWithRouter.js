import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;