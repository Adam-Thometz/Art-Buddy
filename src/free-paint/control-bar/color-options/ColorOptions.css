.ColorOptions {
  background-color: white;
  border-radius: 20px;
  padding-left: 50px;
  transform: translateX(-50px);
  box-shadow: 0 5px 5px rgb(177, 177, 177);
  z-index: 5;
  height: 90px;
  display: flex;
  align-items: center;
}

.ColorOptions-option {
  height: 50px;
  width: 50px;
  border-radius: 50%;
  margin: 10px;
  cursor: pointer;
}