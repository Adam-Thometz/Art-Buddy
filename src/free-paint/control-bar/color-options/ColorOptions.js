import React from 'react';

import { useDispatch } from 'react-redux';
import { setColor } from '_redux/free-paint/freePaintActions';

import './ColorOptions.css';

import paintColors from '_data/free-paint/colors';

const ColorOptions = () => {
  const dispatch = useDispatch();

  const handleChangeColor = e => dispatch(setColor(e.target.id));

  return (
    <div className='ColorOptions'>
      {paintColors.map(color => (
        <div
          className='ColorOptions-option'
          data-testid={`color-option-${color}`}
          id={color}
          style={{ backgroundColor: color }}
          onClick={handleChangeColor}
        ></div>
      ))}
    </div>
  );
};

export default ColorOptions;