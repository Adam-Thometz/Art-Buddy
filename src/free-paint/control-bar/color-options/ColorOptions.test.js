import ColorOptions from "./ColorOptions";

import { render } from "_testUtils/render";

describe("ColorOptions component", () => {
  it("renders without crashing", () => {
    render(<ColorOptions />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<ColorOptions />);
    expect(asFragment()).toMatchSnapshot();
  });
});
