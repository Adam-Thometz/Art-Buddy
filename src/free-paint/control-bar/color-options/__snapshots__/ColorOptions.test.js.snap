// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorOptions component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ColorOptions"
  >
    <div
      class="ColorOptions-option"
      data-testid="color-option-#000000"
      id="#000000"
      style="background-color: rgb(0, 0, 0);"
    />
    <div
      class="ColorOptions-option"
      data-testid="color-option-#FF69B4"
      id="#FF69B4"
      style="background-color: rgb(255, 105, 180);"
    />
    <div
      class="ColorOptions-option"
      data-testid="color-option-#FF0000"
      id="#FF0000"
      style="background-color: rgb(255, 0, 0);"
    />
    <div
      class="ColorOptions-option"
      data-testid="color-option-#FFA500"
      id="#FFA500"
      style="background-color: rgb(255, 165, 0);"
    />
    <div
      class="ColorOptions-option"
      data-testid="color-option-#FFFF00"
      id="#FFFF00"
      style="background-color: rgb(255, 255, 0);"
    />
    <div
      class="ColorOptions-option"
      data-testid="color-option-#00FF00"
      id="#00FF00"
      style="background-color: rgb(0, 255, 0);"
    />
    <div
      class="ColorOptions-option"
      data-testid="color-option-#0000FF"
      id="#0000FF"
      style="background-color: rgb(0, 0, 255);"
    />
    <div
      class="ColorOptions-option"
      data-testid="color-option-#A020F0"
      id="#A020F0"
      style="background-color: rgb(160, 32, 240);"
    />
    <div
      class="ColorOptions-option"
      data-testid="color-option-#964B00"
      id="#964B00"
      style="background-color: rgb(150, 75, 0);"
    />
    <div
      class="ColorOptions-option"
      data-testid="color-option-#808080"
      id="#808080"
      style="background-color: rgb(128, 128, 128);"
    />
  </div>
</DocumentFragment>
`;
